### ISSUE TITLE
Move test selection from make.inc to tests/Makefile

### ISSUE THREAD

**Issue by mathiaswagner** [lattice/quda#265]
I noticed that the selection which tests will be compiled is currently done in `make.inc`, e.g.
```
ifeq ($(strip $(BUILD_HISQ_FORCE)), yes)
  NVCCOPT += -DGPU_HISQ_FORCE -DGPU_STAGGERED_OPROD -DGPU_GAUGE_TOOLS
  COPT += -DGPU_HISQ_FORCE -DGPU_STAGGERED_OPROD -DGPU_GAUGE_TOOLS
  HISQ_PATHS_FORCE_TEST=hisq_paths_force_test
  HISQ_UNITARIZE_FORCE_TEST=hisq_unitarize_force_test
endif
```

I found that to be confusing and would rather have them in `tests/Makefile`.
Since this includes the `make.inc` anyway we can do the selection there as well.

Is that reasonable?

**Comment by mikeaclark** [comment]
This does sound reasonable.  Feel free to do this, it will clean up make.inc.in a bit which is nice.

**Comment by mathiaswagner** [comment]
@AlexVaq I noticed some contract test related variables in `make.inc.in`.

```
ifeq ($(strip $(BUILD_CONTRACT)), yes)
  LOOPVV_TEST=vvTrickLoops
  LOOPVV_TUNE=tuneOneEndTrick
  LOOPTD_TEST=tDilutionLoops
  LPTDHP_TEST=tDilHPELoops
  LOOPCV_TEST=loopsPlain
  LOOPHP_TEST=loopsHPE
  NVCCOPT += -DGPU_CONTRACT
  COPT += -DGPU_CONTRACT
endif
```

I wanted to move all test related stuff to `tests/Makefile`. 
I could not find any references to these variables. So, I would like to remove them. If you need them in some branch please add them there. Or is there a good argument for having them in `make.inc.in` for everyone?


i

**Comment by AlexVaq** [comment]
Those were tests related to the inplementation of interfaces for the contractions, but as we discussed in the last call, they were not very elegant and I never pushed them to any branch (although we use them for production). I think you can remove them safely, but the flag GPU_CONTRACT must stay in COPT and NVCOPT.

**Comment by mathiaswagner** [comment]
Thanks. I did not intend to remove the `-DGPU_CONTRACT`. I only cared about everything suffixed with *_TEST*.

**Comment by mathiaswagner** [comment]
Closed with #266
